airflow-core/src/airflow/api_fastapi/common/dagbag.py

```diff
@@ -14,22 +14,32 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-
 from __future__ import annotations
 
 import os
 from typing import Annotated
 
-from fastapi import Depends
+from fastapi import Depends, Request
 
 from airflow.models.dagbag import DagBag
 from airflow.settings import DAGS_FOLDER
 
 
-def _get_dag_bag() -> DagBag:
+def get_dag_bag() -> DagBag:
+    """Instantiate the appropriate DagBag based on the ``SKIP_DAGS_PARSING`` environment variable."""
     if os.environ.get("SKIP_DAGS_PARSING") == "True":
         return DagBag(os.devnull, include_examples=False)
     return DagBag(DAGS_FOLDER, read_dags_from_db=True)
```

Comment:
Waaaait. Webserver should never parse DAGs. What's going on here?

Reply:
read_dags_from_db is set to True. So, its not actually parsing DAGs, right?

Reply:
Yeah, we don't parse dags in this function.

Reply:
The `SKIP_DAGS_PARSING` is an artifact of old code, will nuke it

Reply:
Done